### PR TITLE
Distributed training setup

### PR DESCRIPTION
**Pull request by MicPie**
PR for the distributed training setup.

### CODE REVIEW

**Comment by MicPie** [comment]
Packages I currently work on:
- grad cache
- pytorch AMP FP16 training

Other packages that will be needed:
- zero optimizer
- review and check (web)dataset setup incl. text mask output and validation dataset
- add accuracy logging
- add ImageNet eval
- test horovod training if needed
- test deep speed training if needed
- see small TO DOs in the code base
